Functions/Statistics/xASL_stat_RobustMean.m

```diff
@@ -50,40 +49,38 @@
 end
 
 
-%% THIS IS PROBABLY REDUNDANT CODE
-% if prod(size(IM))==1 % assume this is a memory mapping file
-%     IM = shiftdim(IM.Data.data,1);
-% end
-% 
-% IM2 = xASL_im_IM2Column(IM,x.S.masks.WBmask); % Make sure to convert to column if too large
-
 %% Compute median, MAD, & deviations
 
-Size4 = size(IM,2);
+Size4 = size(IM, 2);
 
 % Create template image, to compare with
-fprintf('%s\n',['Detecting outliers for n=' num2str(Size4)]);
+fprintf('%s\n',['QC: detecting outliers for n=' num2str(Size4)]);
 
-MedianIM = repmat(xASL_stat_MedianNan(IM,2),[1 Size4]);
+MedianIM = repmat(xASL_stat_MedianNan(IM, 2), [1 Size4]);
 
-if strcmp(ParameterFunction,'SoS')
-     ValueMask = abs(IM - MedianIM);
-elseif strcmp(ParameterFunction,'AI')
-     ValueMask = abs(IM - repmat(IMtemp,[1 Size4])) ./ (0.5.*(IM + repmat(IMtemp,[1 Size4]))); % weighted SoS, AI
-end        
+if strcmpi(ParameterFunction,'SoS')
+     DiffIm = (IM - MedianIM).^2;
+     Deviation = xASL_stat_MeanNan(DiffIm, 1); % gives deviation sum per image, higher is worse quality
```

Comment:
It calculates it along the first dimension. If the input is [X,Y,Z] then the output will not be a scalar, but [1, Y, Z]. 
Is this what you really want?!

Reply:
Yes, the input is the column/vector image (masked), not a 3D image, for efficiency